consumer/consumer.go

```diff
@@ -102,17 +100,24 @@ func (consumer Impl) Start() error {
 	consumed := 0
 	for {
 		msg := <-consumer.PartitionConsumer.Messages()
-		log.Printf("Consumed message offset %d\n", msg.Offset)
+		consumer.ProcessMessage(msg)
 		consumed++
-		orgID, clusterName, report, err := parseMessage(msg.Value)
-		log.Println(orgID, clusterName, report, err)
-		if err != nil {
-			log.Println("Error parsing message from Kafka:", err)
-		}
-		err = consumer.Storage.WriteReportForCluster(orgID, clusterName, report)
-		if err != nil {
-			log.Println("Error writing report to database:", err)
-		}
+	}
+}
+
+// ProcessMessage processes an incoming message
+func (consumer Impl) ProcessMessage(msg *sarama.ConsumerMessage) {
```

Comment:
As totally newbie point of view:
Why don't returning an `error` too? When an `err` is returned from the parsing of the message, you can "inform" about it to the caller

Reply:
yes, we can do it. thanks for review!

Reply:
:-) Glad to help